src/rattler_build_conda_compat/loader.py

```diff
@@ -1,19 +1,29 @@
+from __future__ import annotations
+
 from contextlib import contextmanager
+from typing import TYPE_CHECKING, Any, Iterator, Self
+
 import yaml
-from typing import Any, Dict, Union
+
+from rattler_build_conda_compat.conditional_list import visit_conditional_list
+
+if TYPE_CHECKING:
+    from os import PathLike
 
 
 class RecipeLoader(yaml.BaseLoader):
     @classmethod
     @contextmanager
-    def with_namespace(cls, namespace):
+    def with_namespace(cls: Self, namespace: dict[str, Any] | None) -> Iterator[None]:
         try:
             cls._namespace = namespace
             yield
         finally:
             del cls._namespace
 
-    def construct_sequence(self, node: Any, deep: bool = False) -> Any:
+    def construct_sequence(  # noqa: C901
+        self: RecipeLoader, node: yaml.Node, deep: bool = False  # noqa: FBT002, FBT001
```

Comment:
I think for `self` we can use `Self`

Reply:
I haven't seen a rule written anywhere, but I all examples I can find don't add a type hint to `self` at all.

Reply:
I think there are some examples https://github.com/dry-python/returns/blob/63df68d289115a7414e97cb4a90c47924611fdd5/returns/interfaces/altable.py#L81
we already put the type for `cls`, so I think we should have one "standard" here